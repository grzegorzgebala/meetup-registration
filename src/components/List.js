import React, {Component} from 'react';
import axios from 'axios';

class List extends Component {
	constructor(props) {
        super(props);

        this.state = {
            users: []
        };
    }

    componentDidMount() {
        axios.get('http://localhost:4000/api/productsList')
            .then( (response) => {
                this.setState({users: response.data});
            })
    }

    render() {
        return (
            <div>
                <ul className="list"> 
                    <h3>Users List</h3>
                    <strong className="listElement col-3">Name</strong>
                    <strong className="listElement col-3">Surname</strong>
                    <strong className="listElement col-3">Mail</strong>
                    <strong className="listElement col-2">Date</strong>
                    {	
                        this.state.users.map(
                            (user, index) => <li key={index}>
                            			<div className="listElement col-3">{ user.name }</div>
                            			<div className="listElement col-3">{ user.surname }</div>
                            			<div className="listElement col-3">{ user.mail }</div>
                            			<div className="listElement col-2">{ new Date(user.date).toISOString().slice(0,10) }</div>
                            		</li>
                        )
                    }
                </ul>
            </div>
        );
    }
};

export default List;