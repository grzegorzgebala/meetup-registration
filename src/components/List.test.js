import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

it('renders without crashing', () => {
	const wrapper = shallow(
		<List />
	);
	expect(wrapper).toMatchSnapshot();
});