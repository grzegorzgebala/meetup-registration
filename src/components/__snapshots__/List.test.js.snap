// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <ul
    className="list"
  >
    <h3>
      Users List
    </h3>
    <strong
      className="listElement col-3"
    >
      Name
    </strong>
    <strong
      className="listElement col-3"
    >
      Surname
    </strong>
    <strong
      className="listElement col-3"
    >
      Mail
    </strong>
    <strong
      className="listElement col-2"
    >
      Date
    </strong>
  </ul>
</div>
`;
